RadixWallet/Clients/OnLedgerEntitiesClient/Helpers/OnLedgerEntitiesClient+ComplexTypes.swift

```diff
@@ -147,38 +175,41 @@ extension OnLedgerEntitiesClient {
 
 	private func liquidStakeUnit(
 		_ resource: OnLedgerEntity.Resource,
-		amount: Decimal192,
+		amount: ResourceAmount,
 		validator: OnLedgerEntity.Validator,
 		validatorStakes: [TrackedValidatorStake] = [],
 		guarantee: TransactionGuarantee?
-	) async throws -> ResourceBalance {
-		let worth: Decimal192
+	) async throws -> KnownResourceBalance {
+		let worth: ResourceAmount
 		if validatorStakes.isEmpty {
 			guard let totalSupply = resource.totalSupply, totalSupply.isPositive else {
 				throw MissingPositiveTotalSupply()
 			}
 
-			worth = amount * validator.xrdVaultBalance / totalSupply
+			worth = amount.adjustedNominalAmount { $0 * validator.xrdVaultBalance / totalSupply }
```

Comment:
token arithmetic ought best be done in Sargon, would that makes sense? And with unit tests...